packages/squiggle-lang/src/rescript/Reducer/Reducer_Expression/Reducer_Expression.resi

```diff
@@ -0,0 +1,33 @@
+module Result = Belt.Result
+module T = Reducer_Expression_T
+type expression = T.expression
+type expressionValue = ReducerInterface.ExpressionValue.expressionValue
+type t = expression
+let show: T.expression => Js.String.t
+let showResult: result<T.expression, 'a> => string
+let parse_: (
+  string,
+  string => Result.t<'a, Reducer_ErrorValue.t>,
+  'a => Result.t<expression, Reducer_ErrorValue.t>,
+) => result<expression, Reducer_ErrorValue.t>
+let parse: string => result<expression, Reducer_ErrorValue.t>
+module MapString = Belt.Map.String
+type bindings = MapString.t<unit>
+let defaultBindings: bindings
+let reduceValueList: list<expressionValue> => result<
+  expressionValue,
+  Reducer_ErrorValue.t,
+>
+let reduceExpression: (expression, 'a) => result<
+  expressionValue,
+  Reducer_ErrorValue.t,
+>
+let evalWBindingsExpression: (expression, 'a) => result<
+  expressionValue,
+  Reducer_ErrorValue.t,
+>
+let evalWBindings: (string, bindings) => Result.t<
+  expressionValue,
+  Reducer_ErrorValue.t,
+>
+let eval: string => Result.t<expressionValue, Reducer_ErrorValue.t>
```

Comment:
Do we really need all of these functions? The interface describes the public API. I would have thought that some methods here, like parse_, are basically private.

Reply:
RESOLVED.

Removed parse_

The interface for usage is Reducer.resi. It is not Reducer_Expression.resi .  Here, in this is implementation, I am not supplying a bunch of useful modules under a namespace. All the inner modules are really inner work. Unfortunately, we need the friends keyword of C++. I want to make some functions public but it is for inside use...

Reply:
When you say, "Removed parse_", I don't see this in a commit. Do you have a commit you're working on that will go in, or did you mean something else by this?

I think I'm ready for this code to be merged otherwise, but if you're still working on it, don't want to do that.